samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt

```diff
@@ -34,13 +35,21 @@ open class DefaultComponents(private val applicationContext: Context) {
     val sessionStorage by lazy { DefaultSessionStorage(applicationContext) }
 
     val sessionManager by lazy {
-        SessionManager(engine).apply {
-            sessionStorage.restore(this)
+        SessionManager(engine,
+                initialSessions = restoredSessionsOrInitial,
+                defaultSession = { Session("about:blank") }
+        )
+    }
+
+    private val restoredSessionsOrInitial by lazy {
+        val restored = sessionStorage.persistedSnapshot(engine)
 
-            if (size == 0) {
-                val initialSession = Session("https://www.mozilla.org")
-                add(initialSession)
-            }
+        // If we couldn't read persisted sessions, or there were none, provide an initial session.
```

Comment:
I like the non-side-effecty persist/restore but I think this we just made harder to crasp. How about we change this to:

```Kotlin
SessionManager(engine, 
  initialSessions = sessionStorage.restore(engine), 
  defaultSession = { Session("about:blank") }
).apply {
  if (size == 0) {
    add(Session("https://www.mozilla.org"))
  }
}
```

Reply:
I should clarify my concern here. It's basically that we put 3 new concepts in front of the API user. `initialSession`, `defaultSession` and `SessionsSnapshot`. So, I was looking for a way to hide `SessionsSnapshot`...